plugin/neomake.vim

```diff
@@ -1,37 +1,11 @@
 " vim: ts=4 sw=4 et
 
-function! s:NeomakeCommand(file_mode, enabled_makers)
-    if a:file_mode
-        call neomake#Make({
-            \ 'enabled_makers': len(a:enabled_makers) ?
-                \ a:enabled_makers :
-                \ neomake#GetEnabledMakers(&ft),
-            \ 'ft': &ft,
-            \ 'file_mode': 1,
-            \ })
-    else
-        call neomake#Make({
-            \ 'enabled_makers': len(a:enabled_makers) ?
-                \ a:enabled_makers :
-                \ neomake#GetEnabledMakers()
-            \ })
-    endif
-endfunction
-
-function! s:NeomakeSh(sh_command)
-    let custom_maker = neomake#utils#MakerFromCommand(&shell, a:sh_command)
-    let custom_maker.name = 'sh: '.a:sh_command
-    let custom_maker.remove_invalid_entries = 0
-    let enabled_makers =  [custom_maker]
-    call neomake#Make({'enabled_makers': enabled_makers})
-endfunction
-
-command! -nargs=* -bang -bar -complete=customlist,neomake#CompleteMakers Neomake call s:NeomakeCommand(<bang>1, [<f-args>])
+command! -nargs=* -bang -bar -complete=customlist,neomake#CompleteMakers Neomake call neomake#WithCallback(<bang>1, [<f-args>])
```

Comment:
I think it won't work to pass the callback in the command here. Right now, `:Neomake` only takes a list of commands as arguments and we shouldn't break that expectation. Instead, make it configurable like this:

```
let g:neomake_python_pylint_exit_callback = 'MyExitCallback'
```

Add `exit_callback` to the `defaults` dict to get this working: https://github.com/benekastah/neomake/pull/162/files#diff-1ceaba5126b0cf477e72f87cb2767af0L150